src/main/java/dev/mccue/resolve/maven/PomParser.java

```diff
@@ -11,7 +11,7 @@
 import java.util.function.Function;
 import java.util.stream.Collectors;
 
-public final class PomParser extends DefaultHandler {
+public final class PomParser extends DefaultHandler { //TODO extends DefaultHandler -> Implements SaxParser
```

Comment:
The interface you should abstract over is probably just `ContentHandler`. This code wouldn't change, but the code that wants to use what coursier calls a SaxParser probably just needs to require that.